src/main/kotlin/com/cognifide/gradle/aem/common/instance/service/osgi/OsgiFramework.kt

```diff
@@ -141,10 +191,33 @@ class OsgiFramework(sync: InstanceSync) : InstanceService(sync) {
             aem.logger.info("Triggering OSGi framework shutdown on $instance.")
             sync.http.postUrlencoded(VMSTAT_PATH, mapOf("shutdown_type" to type))
         } catch (e: AemException) {
-            throw InstanceException("Cannot trigger shutdown of $instance.", e)
+            throw InstanceException("Cannot trigger shutdown of $instance. Cause: ${e.message}", e)
         }
     }
 
+    private fun getConfigurationsList(): List<ConfigurationState.Pid> {
+        return try {
+            sync.http.get(CONFIGURATION_PATH) { response ->
+                val html = asString(response)
+                val configJson = CONFIGURATIONS_REGEX.find(html)?.groups?.get(1)?.value ?: "{}"
+                Formats.jsonMapper(false).readValue(configJson, ConfigurationState::class.java)
+            }
+        } catch (e: AemException) {
+            throw InstanceException("OSGi configuration list cannot be fetched from $instance. Cause: ${e.message}", e)
```

Comment:
how about catching `RequestException` and separately `ResponseException` to distinguish parse error and request error? 

as in https://github.com/Cognifide/gradle-aem-plugin/blob/master/src/main/kotlin/com/cognifide/gradle/aem/common/instance/service/pkg/PackageManager.kt#L87-L91

this gives us previse error reporting; we know that instance request failed (e.g have client error) or know that GAP cannot parse some wrong instance response

Reply:
this is core method of your feature, it would be nice to have good error reporting here (this could potentially fail most quickly)

Reply:
```kotlin
val configJson = CONFIGURATIONS_REGEX.find(html)?.groups?.get(1)?.value
    ?: throw ResponseException("OSGi configuration cannot be find in console response of $instance.")
```

Reply:
currently we will have false-postive without it; e.g when they change `configData` var in felix webconsole